### PR TITLE
🤝 PIC-2420: Create contract for POST hearing to court-case-service

### PR DESCRIPTION
**Pull request by jhevans**
Proposed contract for POSTing and GETing hearing comments from `court-case-service`

### CODE REVIEW

File: tests/pact/case-service/create-hearing-comment.test.pact.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    })
+
+    it('Creates hearing comment successfully', async () => {
+      const addCommentResponse = {
+        ...requestBody,
+        created: '2022-08-24T15:42:41',
+        createdByUuid: 'cc2285f2-91b0-4e00-bd5e-9bdc35896bb6',
+        commentId: 12234
+      }
+      await provider.addInteraction({
```
Comment (jhevans) [inline tests/pact/case-service/create-hearing-comment.test.pact.js:31]: This is the contract

**Review by vb-moj — APPROVED** [review]

File: schemas/post-hearing-comment.schema.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      "type": "string",
+      "title": "The hearing id",
+      "default": ""
+    },
+    "createdByUuid": {
+      "$id": "#/properties/caseComments/items/anyOf/0/properties/createdByUuid",
+      "type": "string",
+      "title": "The created user uuid",
+      "default": ""
+    },
```
Comment (sarahyjja) [inline schemas/post-hearing-comment.schema.json:36]: I was wondering, would it be good to make the "createdByUuid" as a `required` parameter too?

**Review by rschallapalli — CHANGES_REQUESTED** [review]
LGTM  (minor: there are some references to caseComments and caseId to be replaced with hearing) .. ignore me if this is still a draft

**Review by rschallapalli — COMMENTED** [review]
In the hearing / case-progress area we are calling these as notes instead of comments so we may want to align the naming to notes?

File: schemas/post-hearing-comment.schema.json (hunk 2 above)
Comment (jhevans) [reply schemas/post-hearing-comment.schema.json:36]: Fair shout, although @rschallapalli do we now need this at all? If it gets this from the user token then we can drop this entirely.

**Comment by jhevans** [comment]
Yes good point - they're notes not comments! I will amend!

File: schemas/post-hearing-comment.schema.json (hunk 2 above)
Comment (rschallapalli) [reply schemas/post-hearing-comment.schema.json:36]: No, we don't need any apart from author, comment, caseId

Comment (jhevans) [reply schemas/post-hearing-comment.schema.json:36]: Interesting - this schema defines the response, not the request. We're not checking that as such except for the Pact itself...

Comment (jhevans) [reply schemas/post-hearing-comment.schema.json:36]: It turns out that schema is serving as dual purpose for both the request and response with different fields appearing in each - I'll split them out to make the distinction clear

**Pull request closed without merging**